### ISSUE TITLE
getFragmentManager() always returns null

### ISSUE THREAD

**Issue by codeanticode** [processing/processing-android#507]
The value getActivity().getFragmentManager() is not returned by PApplet.getFragmentManager().

**Comment by codeanticode** [comment]
Fixed with https://github.com/processing/processing-android/commit/8900bc38caed51d71f1ec21b1182e3b37c563a83

**Comment by HansAMpaD** [comment]
You are really fast. Thanks again for all the work, appreciate it very much!

**Comment by HansAMpaD** [comment]
It builds but fails at start:

BUILD SUCCESSFUL in 11s
28 actionable tasks: 28 executed
 Blepdroid starting Blepdroid(PApplet _parent) 
FATAL EXCEPTION: Animation Thread
Process: processing.test.ampad_wind_10_android, PID: 20154
java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.FragmentTransaction android.app.FragmentManager.beginTransaction()' on a null object reference
	at blepdroid.Blepdroid.<init>(Blepdroid.java:290)
	at processing.test.ampad_wind_10_android.AMpaD_Wind_10_android.setup(AMpaD_Wind_10_android.java:232)
	at processing.core.PApplet.handleDraw(PApplet.java:1846)
	at processing.core.PSurfaceNone.callDraw(PSurfaceNone.java:476)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:516)

**Comment by HansAMpaD** [comment]
running scan
calling startLeScan
device discovered 
discovered device Adafruit Bluefruit LE address: CD:2E:55:09:1B:4F rssi: -68
 trying to connect Adafruit Bluefruit LE 
 connectDevice 
 connectDevice::connected device 
 BluetoothLEService connectDevice  
**This is were it stops:**
java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at blepdroid.Blepdroid.addDevice(Blepdroid.java:625)
	at blepdroid.BlepdroidAdapter$2.onScanResult(BlepdroidAdapter.java:97)
	at android.bluetooth.le.BluetoothLeScanner$BleScanCallbackWrapper$1.run(BluetoothLeScanner.java:463)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6823)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)
Caused by: java.lang.NoSuchMethodError: No virtual method keySet()Ljava/util/concurrent/ConcurrentHashMap$KeySetView; in class Ljava/util/concurrent/ConcurrentHashMap; or its super classes (declaration of 'java.util.concurrent.ConcurrentHashMap' appears in /system/framework/core-libart.jar)
	at blepdroid.BluetoothLeService.connect(BluetoothLeService.java:150)
	at blepdroid.Blepdroid.connectDevice(Blepdroid.java:350)
	at info.ampad.AMpaD_Comp5.AMpaD_Wind_101_android.onDeviceDiscovered(AMpaD_Wind_101_android.java:2264)
	... 11 more
 scanDevices 
 check functionality

**Comment by codeanticode** [comment]
@HansAMpaD Make sure to use version 271 (v4.0.4) of the android mode, it should be available through the Contributions Manager in the PDE, you can also download it manually from [here](https://github.com/processing/processing-android/releases/tag/android-271). Are you using [this package](https://github.com/codeanticode/blepdroid/releases) for the blepdroid library?

**Comment by HansAMpaD** [comment]
I installed 271 manually and replaced the Blepdroid files. After updating the sketch and running it, previous problems where solved. It finds the BLE devices but then it gives the error and it won’t actually connect with the device.

 

Caused by: java.lang.NoSuchMethodError: No virtual method keySet()Ljava/util/concurrent/ConcurrentHashMap$KeySetView; in class Ljava/util/concurrent/ConcurrentHashMap; or its super classes (declaration of 'java.util.concurrent.ConcurrentHashMap' appears in /system/framework/core-libart.jar)

  

This is what I found with Google: “the problem is that they have the same package & class names, when you compile, one of the class is overriding the other while your IDE is thinking you are using the class which gets overridden”

 

Of course not sure if this is the case, but sounds kind of logical. 

 

 

Van: codeanticode <notifications@github.com>
Beantwoorden - Aan: processing/processing-android <reply@reply.github.com>
Datum: zaterdag 29 december 2018 om 08:42
Aan: processing/processing-android <processing-android@noreply.github.com>
CC: HansAMpaD <jhvos01@gmail.com>, Mention <mention@noreply.github.com>
Onderwerp: Re: [processing/processing-android] getFragmentManager() always returns null (#507)

 

@HansAMpaD Make sure to use version 271 (v4.0.4) of the android mode, it should be available through the Contributions Manager in the PDE, you can also download it manually from here. Are you using this package for the blepdroid library?

**Comment by codeanticode** [comment]
@HansAMpaD I tried to reproduce your error but was unable to do so. I don't have any BLE device that exposes services, which seems that the functionality that is giving you trouble. But I created other hash maps in my app, and did not get the virtual method error when calling getKeys(). I think the best route would be to open an issue in the [blepdroid repo](https://github.com/joshuajnoble/blepdroid/issues).
